### PR TITLE
Fix multi user notebook

### PR DESCRIPTION
**Pull request by akhenry**
Closes https://github.com/nasa/openmct/issues/5562

### Describe your changes:
1. Listens for any changes to the notebook object to detect addition, removal, and modification of entries, pages, and sections. Previously this was not necessary because the notebook object was a MutableDomainObject and the entries were a computed property that was automatically re-computed when the MutableDomainObject was updated as a result of a remote change to the notebook. Since the entries are no longer a computed property it is necessary to listen for changes to the domain object "manually".
2. In the process of fixing this it was discovered that changes to notebook entry tags were also not automatically being reflected to all users in real-time, even after fixing the regression. This was due to the fact that the tags are stored on an Annotation object, and not the notebook itself, thus no modification was occurring and the notebook entries were not being re-rendered. This PR adds `modified` and `modifiedBy` attributes to notebook entries, mirroring the `created` and `createdBy` attributes to track modifications to entries. In doing so it is also possible to detect the creation of new annotations without having to listen to all possible annotations on an object.
3. Fixes a bug in the existing test specs that prevented this regression from being caught automatically. Previously all of the tests that test for remote modifications to notebooks were invalid due to sharing an object instance.
### All Submissions:

* [ ] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [ ] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate unit tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Commit messages meet standards?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [x] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by unlikelyzero** [comment]
@akhenry @scottbell can we add a multi user notebook test to cover this basic scenario?

File: src/plugins/notebook/components/NotebookEntry.vue [hunk 1]
```diff
@@ -333,6 +335,17 @@ export default {
                 return found;
             });
 
+            this.timestampAndUpdate();
+        },
+        async timestampAndUpdate() {
+            const user = await this.openmct.user.getCurrentUser();
+
+            if (user !== undefined) {
```
Comment (jvigliotta) [inline src/plugins/notebook/components/NotebookEntry.vue:343]: I think you want this to be `if (user === undefined)` ?

File: src/plugins/notebook/components/NotebookEntry.vue [hunk 2]
```diff
@@ -333,6 +335,17 @@ export default {
...
+                this.entry.modifiedBy = 'Unknown';
```
Comment (scottbell) [inline src/plugins/notebook/components/NotebookEntry.vue:344]: Is this a magic string we'll need to use elsewhere?

File: src/plugins/notebook/pluginSpec.js [hunk 3]
```diff
@@ -211,10 +211,17 @@ describe("Notebook plugin:", () => {
 
         describe("synchronization", () => {
 
+            let objectCloneToSyncFrom;
+
+            beforeEach(() => {
+                objectCloneToSyncFrom = JSON.parse(JSON.stringify(notebookViewObject));
```
Comment (scottbell) [inline src/plugins/notebook/pluginSpec.js:217]: Do we want to use [structuredClone](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone) here? It looks pretty well supported now.

**Review by scottbell — COMMENTED** [review]
Some small comments, but the annotation/tagging stuff looks great!

**Comment by akhenry** [comment]
> @akhenry @scottbell can we add a multi user notebook test to cover this basic scenario?

There are already Jasmine tests that would have caught this regression except they had a bug, which I've now fixed, and added to for some additional edge cases.

I think adding e2e tests for this regression would be redundant at this point. Due to the nature of multi-user notebooks, they're actually really hard to test properly as well. I'm not convinced e2e tests can add much value here.

File: src/plugins/notebook/components/NotebookEntry.vue (hunk 1 above)
Comment (akhenry) [reply src/plugins/notebook/components/NotebookEntry.vue:343]: Ah, good catch, thanks.

File: src/plugins/notebook/components/NotebookEntry.vue (hunk 2 above)
Comment (akhenry) [reply src/plugins/notebook/components/NotebookEntry.vue:344]: It's a magic value, yup. I'll make it a const :) Not sure if we will ever need it anywhere else.

File: src/plugins/notebook/pluginSpec.js (hunk 3 above)
Comment (akhenry) [reply src/plugins/notebook/pluginSpec.js:217]: Was unaware of this function, cool, thanks! I'll change it.

**Comment by akhenry** [comment]
@scottbell @jvigliotta  Believe I've addressed everything, thank you!

**Comment by scottbell** [comment]
Looking good @akhenry!

https://user-images.githubusercontent.com/9853862/181726456-db079d65-324e-4f55-bf19-fa0cc9b85182.mov

**Review by scottbell — APPROVED** [review]
